sopel/irc/backends.py

```diff
@@ -389,16 +391,78 @@ async def _run_forever(self) -> None:
         # open connection
         try:
             self._reader, self._writer = await asyncio.open_connection(
-                **self.get_connection_kwargs(),
+                **connection_kwargs,
             )
-        except ssl.SSLError:
-            LOGGER.exception('Unable to connect due to SSL error.')
+
+        # SSL Error
+        except ssl.SSLCertVerificationError as err:
+            LOGGER.error(
+                'Unable to connect due to an SSL validation error: %s',
+                err,
+            )
+            self.log_exception()
+            # tell the bot to quit without restart
+            self.bot.hasquit = True
+            self.bot.wantsrestart = False
+            return
+        except ssl.SSLError as err:
+            LOGGER.error('Unable to connect due to an SSL error: %s', err)
+            self.log_exception()
```

Comment:
Additionally, I wonder if it would be helpful to the user #2111 has in mind to pull out the [specific message](https://docs.python.org/3/library/ssl.html#ssl.SSLCertVerificationError.verify_code) associated with the [SSL diagnostic](https://www.openssl.org/docs/man1.0.2/man1/verify.html#DIAGNOSTICS) that a verification error will indicate, i.e. something like:

```python
            LOGGER.error(
                'Unable to connect due to SSL certificate verification failure: %s (%s)',
                err.verify_message,
                err.verify_code,
            )
```

and since the exception is logged anyway, this loses no information

Reply:
IIRC, the `%s` version for the SSLCertVerificationError and the SSLError exceptions already contains that information, but I could be wrong. It's kind hard to test that something is improperly configured for SSL because most server are actually working properly. :grin: